### PR TITLE
Add Esqueleto, beef up Persist

### PR DESCRIPTION
**Pull request by saramuse**
Story details: https://app.shortcut.com/itprotv/story/98309

---

- [ ] I manually tested the code locally to make sure that it works.
- [x] If there is documentation, I made sure it's accurate and up to date.
- [ ] If possible, I wrote automated tests for the new behavior.

### CODE REVIEW

**Review by codygman — APPROVED** [review]

File: Prolude/Core.hs [hunk 1]
```diff
@@ -40,22 +40,28 @@ module Prolude.Core
...
 import Data.Bool (Bool(False, True), not, otherwise, (&&), (||))
 import Data.Bifunctor (Bifunctor(bimap, first, second))
 import Data.Char (Char, chr, ord)
+import Data.Coerce (coerce)
 import Data.Either (Either(Left, Right), either)
 import Data.Eq (Eq((/=), (==)))
 import Data.Foldable
   (Foldable(elem, foldMap, foldr, length, null, sum), all, and, any, concat, concatMap, mapM_, or)
-import Data.Function ((&))
+import Data.Function ((&), (.))
```
Comment (tfausak) [inline Prolude/Core.hs:64]: ... we weren't exporting this before? How is that possible?

File: prolude.cabal [hunk 2]
```diff
@@ -52,6 +53,7 @@ library
         , bytestring >= 0.10.10 && < 0.11
         , cassava >= 0.5.2.0 && < 0.6
         , containers >= 0.6.2 && < 0.7
+        , esqueleto >= 3.5.3.0 && < 3.6.4
```
Comment (tfausak) [inline prolude.cabal:56]:
```suggestion
        , esqueleto >= 3.5.3 && < 3.6
```

File: prolude.cabal [hunk 3]
```diff
@@ -63,9 +65,11 @@ library
         , persistent-mongoDB >= 2.8 && < 3
         , safe-exceptions >= 0.1.7 && < 0.2
         , scientific >= 0.3.6 && < 0.4
-        , servant >= 0.18.2 && < 0.19
+        , servant >= 0.18.1 && < 0.19
+        , stm >= 2.5.0.0 && < 2.6
```
Comment (tfausak) [inline prolude.cabal:69]:
```suggestion
        , stm >= 2.5.0 && < 2.6
```

File: prolude.cabal [hunk 4]
```diff
@@ -63,9 +65,11 @@ library
...
         , swagger2 >= 2.5 && < 2.7
         , text >= 1.2.3 && < 1.3
         , time >= 1.9.3 && < 1.10
         , uuid >= 1.3.13 && < 1.4
         , vector >= 0.12.1 && < 0.13
+        , witch >= 0.3.4.0 && < 0.3.5
```
Comment (tfausak) [inline prolude.cabal:75]:
```suggestion
        , witch >= 0.3.4 && < 0.3
```

**Review by tfausak — APPROVED** [review]

File: Prolude/Core.hs (hunk 1 above)
Comment (codygman) [reply Prolude/Core.hs:64]: Not sure, I know prelude exports it but we hide all of prelude right?

Comment (tfausak) [reply Prolude/Core.hs:64]: Ah, it's part of the `GHC.Base` import:

https://github.com/EdutainmentLIVE/prolude/blob/25388a70dd39b55a3e389ac87baeb404d2076c1f/Prolude/Core.hs#L88

Comment (saramuse) [reply Prolude/Core.hs:64]: My exact thoughts while adding this 😅

File: prolude.cabal [hunk 5]
```diff
@@ -63,9 +65,11 @@ library
...
         , scientific >= 0.3.6 && < 0.4
-        , servant >= 0.18.2 && < 0.19
+        , servant >= 0.18.1 && < 0.19
+        , stm >= 2.5.0 && < 2.6
         , swagger2 >= 2.5 && < 2.7
         , text >= 1.2.3 && < 1.3
         , time >= 1.9.3 && < 1.10
         , uuid >= 1.3.13 && < 1.4
         , vector >= 0.12.1 && < 0.13
+        , witch >= 0.3.4 && < 0.3
```
Comment (tfausak) [inline prolude.cabal:75]: Oops.
```suggestion
        , witch >= 0.3.4 && < 0.4
```

**Pull request merged**